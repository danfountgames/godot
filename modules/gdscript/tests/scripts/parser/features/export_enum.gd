const Utils = preload("../../utils.notest.gd")

@export_enum("Red", "Green", "Blue") var test_untyped

@export_enum("Red", "Green", "Blue") var test_weak_int = 0
@export_enum("Red", "Green", "Blue") var test_weak_string = ""

@export_enum("Red", "Green", "Blue") var test_hard_int: int
@export_enum("Red", "Green", "Blue") var test_hard_string: String

@export_enum("Red:10", "Green:20", "Blue:30") var test_with_values

func test():
	for property in get_property_list():
		if str(property.name).begins_with("test_"):
			Utils.print_property_extended_info(property, self)
