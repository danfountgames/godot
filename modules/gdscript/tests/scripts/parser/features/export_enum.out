GDTEST_OK
var test_untyped: int = null
  hint=ENUM hint_string="Red,Green,Blue" usage=DEFAULT|SCRIPT_VARIABLE
var test_weak_int: int = 0
  hint=ENUM hint_string="Red,Green,Blue" usage=DEFAULT|SCRIPT_VARIABLE
var test_weak_string: String = ""
  hint=ENUM hint_string="Red,Green,Blue" usage=DEFAULT|SCRIPT_VARIABLE
var test_hard_int: int = 0
  hint=ENUM hint_string="Red,Green,Blue" usage=DEFAULT|SCRIPT_VARIABLE
var test_hard_string: String = ""
  hint=ENUM hint_string="Red,Green,Blue" usage=DEFAULT|SCRIPT_VARIABLE
var test_with_values: int = null
  hint=ENUM hint_string="Red:10,Green:20,Blue:30" usage=DEFAULT|SCRIPT_VARIABLE
